pyccel/ast/numpyext.py

```diff
@@ -416,13 +423,35 @@ def Shape(arg):
         return PythonTuple(*arg.shape)
 
 #==============================================================================
+
 class NumpyImag(PythonImag):
-    """Represents a call to  numpy.real for code generation.
+    """Represents a call to  numpy.imag for code generation.
 
     > a = 1+2j
     > np.imag(a)
     2.0
     """
+    def __new__(cls, arg):
+        if not isinstance(arg.dtype, NativeComplex):
+            dtype=NativeInteger() if isinstance(arg.dtype, NativeBool) else arg.dtype
+            if isinstance(arg, Literal) or (isinstance(arg, Variable) and not arg.allocatable):
```

Comment:
Shouldn't this condition be `if isinstance(arg, Literal) and arg.rank == 0` ? This case doesn't handle the  case where the argument is a stack array